opaque_keys/edx/django/models.py

```diff
@@ -190,6 +198,53 @@ class UsageKeyField(OpaqueKeyField):
     """
     description = "A Location object, saved to the DB in the form of a string"
     KEY_CLASS = UsageKey
+    # Default to 191 characters as the maximum length of this field type,
+    # because we want to support the MySQL "utf8mb4" character set (real UTF-8)
+    # and the InnoDB index limit on common MySQL configurations is 767 bytes
+    # (191 characters * 4 bytes/character = 764 bytes)
+    DEFAULT_MAX_LENGTH = 191
+
+
+class UsageKeyCourseLookup(StartsWith):
+    """
+    Allows efficiently querying the course that a UsageKey belongs to,
+    without requiring that the CourseKey is stored in a separate column.
+
+    This only works with UsageKeys that contain the full CourseKey in
+    their serialized form, preceding any usage-specific identifiers.
+
+    Usage: my_model.objects.filter(usage_key__course=course_key)
+    """
+    lookup_name = 'course'
```

Comment:
Nit: Since the property name on the UsageKey is `course_key` would that be a nicer name?